hsmd/libhsmd.c

```diff
@@ -2290,14 +2288,15 @@ u8 *hsmd_init(struct secret hsm_secret,
 		    &secretstuff.hsm_secret, sizeof(secretstuff.hsm_secret),
 		    "derived secrets", strlen("derived secrets"));
 
+	u64 mutual_version = hsmd_version < 5 ? hsmd_version : 5;
```

Comment:
This code will need to change as we progress to 6, 7, ...

I think the following approach won't have to change with future protocol changes:
1. In the calling routine, after the current min/max range checks, `mutual_version = min(node_max_version, hsmd_max_version)`
2. Pass `mutual_version` to this routine instead and simply put in the reply message

Reply:
Ok what do you think of https://github.com/ElementsProject/lightning/pull/7088/commits/3bad94e1252a55de7ba415d563800698a75c3a0c